src/services/StructuralMetadataUtils.test.js

```diff
@@ -0,0 +1,593 @@
+import StructuralMetadataUtils from './StructuralMetadataUtils';
+import {
+  testData,
+  testEmptyHeaderBefore,
+  testEmptyHeaderAfter
+} from '../test/TestStructure';
+
+const smu = new StructuralMetadataUtils();
+
+describe('StructuralMetadataUtils class', () => {
+  test('creates a helper drop zone object for drag and drop', () => {
+    const value = smu.createDropZoneObject();
+    expect(value).toHaveProperty('id');
+    expect(value).toHaveProperty('type', 'optional');
+  });
+
+  test('creates a helper span object', () => {
+    const obj = {
+      beginTime: '00:00:01',
+      endTime: '00:00:02',
+      timespanChildOf: '3bf42620-2321-11e9-aa56-f9563015e266',
+      timespanTitle: 'Tester'
+    };
+    const value = smu.createSpanObject(obj);
+
+    expect(value).toHaveProperty('id');
+    expect(value).toHaveProperty('type', 'span');
+    expect(value).toHaveProperty('begin', '00:00:01');
+    expect(value).toHaveProperty('end', '00:00:02');
+    expect(value).toHaveProperty('label', 'Tester');
+  });
+
+  describe('tests new time overlaps existing time ranges', () => {
+    const allSpans = smu.getItemsOfType('span', testData);
+    test('time == 00:00:00.00 (before the first timespan)', () => {
+      const time = '00:00:00.00';
+      expect(smu.doesTimeOverlap(time, allSpans)).toBeTruthy();
```

Comment:
This is my fault, but the function title `doesTimeOverlap` seems backwards from what the function returns:)  I might change that.

Reply:
@adamjarling Oh yes. When I first wrote this I asserted the values the other way :)